### PR TITLE
[KEYCLOAK-10443] - Define a global decision strategy for resource servers

### PR DESCRIPTION
**Pull request by pedroigor**
Documentation: https://github.com/keycloak/keycloak-documentation/pull/680

### CODE REVIEW

**Comment by pedroigor** [comment]
@mposolda, could you please check if the migration changes look good ?

File: server-spi-private/src/main/java/org/keycloak/models/utils/ModelToRepresentation.java [hunk 1]
```diff
@@ -788,6 +788,7 @@ public static ResourceServerRepresentation toRepresentation(ResourceServer model
         server.setName(client.getClientId());
```
Comment (oridool) [inline server-spi-private/src/main/java/org/keycloak/models/utils/ModelToRepresentation.java]: in the line :
server.setClientId(model.getId());
Shouldn't that be:
server.setClientId(client.getId());
?

**Comment by stianst** [comment]
Migratoin versoin should be 7, not 6.0.2. 6.0.2 if released will not be released from master branch and include new features/enhancements.

**Review by oridool — COMMENTED** [review]
looks good, minor comment/question

**Comment by pedroigor** [comment]
Duhh ... Just look the pom version :) Will fix it.

File: server-spi-private/src/main/java/org/keycloak/models/utils/ModelToRepresentation.java (hunk 1 above)
Comment (pedroigor) [reply server-spi-private/src/main/java/org/keycloak/models/utils/ModelToRepresentation.java]: Both client and RS share the same ID.

**Review by abstractj — APPROVED** [review]

File: model/jpa/src/main/java/org/keycloak/authorization/jpa/entities/ResourceServerEntity.java [hunk 2]
```diff
@@ -66,6 +70,17 @@ public void setPolicyEnforcementMode(PolicyEnforcementMode policyEnforcementMode
         this.policyEnforcementMode = policyEnforcementMode;
     }
 
+    public void setDecisionStrategy(DecisionStrategy decisionStrategy) {
```
Comment (abstractj) [inline model/jpa/src/main/java/org/keycloak/authorization/jpa/entities/ResourceServerEntity.java]: @pedroigor based on the change below. Would make sense to remove CONSENSUS from https://github.com/keycloak/keycloak-documentation/blob/532a46f60f04ef709776d1599359bc9ab24e10a4/authorization_services/topics/policy-aggregated-policy.adoc#decision-strategy-for-aggregated-policies  and https://github.com/keycloak/keycloak-documentation/blob/532a46f60f04ef709776d1599359bc9ab24e10a4/authorization_services/topics/permission-decision-strategy.adoc#policy-decision-strategies ?

File: model/jpa/src/main/java/org/keycloak/authorization/jpa/entities/ResourceServerEntity.java [hunk 3]
```diff
@@ -66,6 +70,17 @@ public void setPolicyEnforcementMode(PolicyEnforcementMode policyEnforcementMode
...
+        if (DecisionStrategy.CONSENSUS.equals(decisionStrategy)) {
+            throw new IllegalArgumentException("Strategy " + decisionStrategy + " not supported");
+        }
+        this.decisionStrategy = decisionStrategy;
+    }
```
Comment (abstractj) [inline model/jpa/src/main/java/org/keycloak/authorization/jpa/entities/ResourceServerEntity.java]: @pedroigor based on the change above. Would make sense to remove CONSENSUS from https://github.com/keycloak/keycloak-documentation/blob/532a46f60f04ef709776d1599359bc9ab24e10a4/authorization_services/topics/policy-aggregated-policy.adoc#decision-strategy-for-aggregated-policies  and https://github.com/keycloak/keycloak-documentation/blob/532a46f60f04ef709776d1599359bc9ab24e10a4/authorization_services/topics/permission-decision-strategy.adoc#policy-decision-strategies ?

Comment (pedroigor) [reply model/jpa/src/main/java/org/keycloak/authorization/jpa/entities/ResourceServerEntity.java]: As per your revision to https://github.com/keycloak/keycloak-documentation/pull/680/files. We are not referencing `CONSENSUS` for the new configuration option.

**Pull request merged**